pkg/controllers/updaterun/controller.go

```diff
@@ -104,11 +104,26 @@ func (r *Reconciler) Reconcile(ctx context.Context, req runtime.Request) (runtim
 	// Emit the update run status metric based on status conditions in the updateRun.
 	defer emitUpdateRunStatusMetric(updateRun)
 
+	// Early check for abandoned state - this is a terminal state, no initialization needed.
+	state := updateRun.GetUpdateRunSpec().State
+	if state == placementv1beta1.StateAbandoned {
+		klog.V(2).InfoS("The updateRun is abandoned, terminating", "state", state, "updateRun", runObjRef)
+		return runtime.Result{}, r.recordUpdateRunAbandoned(ctx, updateRun)
+	} else if state == placementv1beta1.StateStopped { // Early check for stopped state - pause the update run if needed.
+		klog.V(2).InfoS("The updateRun is stopped, waiting to resume", "state", state, "updateRun", runObjRef)
+		return runtime.Result{}, r.recordUpdateRunPaused(ctx, updateRun)
+	}
+
```

Comment:
nit: use switch to replace if/else
```suggestion
	switch state {
	case placementv1beta1.StateAbandoned:
		klog.V(2).InfoS("The updateRun is abandoned, terminating", "state", state, "updateRun", runObjRef)
		return runtime.Result{}, r.recordUpdateRunAbandoned(ctx, updateRun)
	case placementv1beta1.StateStopped: // Early check for stopped state - pause the update run if needed.
		klog.V(2).InfoS("The updateRun is stopped, waiting to resume", "state", state, "updateRun", runObjRef)
		return runtime.Result{}, r.recordUpdateRunPaused(ctx, updateRun)
	}
```